src/sentry/models/monitor.py

```diff
@@ -142,10 +144,20 @@ class Monitor(Model):
     class Meta:
         app_label = "sentry"
         db_table = "sentry_monitor"
-        index_together = (("type", "next_checkin"),)
+        index_together = (("organization_id", "slug"), ("type", "next_checkin"))
 
     __repr__ = sane_repr("guid", "project_id", "name")
 
+    def save(self, *args, **kwargs):
+        # TODO(epurkhsier): This logic is to be removed when the `guid` field
+        # is removed and a slug is required when creating monitors
+        if self.slug is None:
+            self.guid = uuid4()
```

Comment:
Actually, this is dangerous

Reply:
We need to make sure this only happens during creation, otherwise there is a scenario where the model is updated and the slug is null, so we would generate a NEW guid. Not good

Reply:
I've fixed this by checking `self._state.adding is True` and adding a notice why we're doing that.